### PR TITLE
Added AZ filter to machinepool subnet filters

### PR DESCRIPTION
**Pull request by AverageMarcus**
Signed-off-by: Marcus Noble <github@marcusnoble.co.uk>

### CODE REVIEW

File: helm/cluster-aws/templates/_machine_pools.tpl [hunk 1]
```diff
@@ -45,6 +45,8 @@ spec:
     - name: tag:sigs.k8s.io/cluster-api-provider-aws/role
       values:
       - private
+    - name: availabilityZone
+      values: {{ .availabilityZones | default ( include "aws-availability-zones" .) }}
```
Comment (calvix) [inline helm/cluster-aws/templates/_machine_pools.tpl:49]: this is not a tag but a property?

Comment (calvix) [reply helm/cluster-aws/templates/_machine_pools.tpl:49]: does it properly template with one or multiple AZs?

**Review by mnitchev — APPROVED** [review]
Don't have much context on aws filters, but looks good otherwise.

File: helm/cluster-aws/templates/_machine_pools.tpl (hunk 1 above)
Comment (AverageMarcus) [reply helm/cluster-aws/templates/_machine_pools.tpl:49]: Yes, to both.

**Pull request merged**